### DISCUSSION TITLE
Azure and Application Insights?

### DISCUSSION THREAD

**Discussion by plettb** [CoreWCF/CoreWCF#670] [Q&A] [answered]
This seems bizarre to me, but... if I publish a CoreWCF app to Azure and then enable Application Insights, the app stops responding and produces only 502 and 503 errors.  There is no obvious connection in my mind, but it really is as simple as this:  Application Insights enabled --> app fails; Application Insights disabled --> app runs.  I can actually toggle it back and forth and the app stops and/or starts working apparently entirely dependent on whether AI is enabled.  I do absolutely no configuration of logging, just using the defaults as supplied by the environment, and simply call app.Logger.LogInformation.

Has anyone else run into this?  Any insight?

Thanks!

**Comment by mconnew** [comment, marked as answer]
Your timing couldn't have been better. I spent the end of last week investigating this and discovered a bug in the AI Azure extensions implementation which hijacks one of the application lifetime event mechanisms in asp.net core and effectively disables it. I just pushed a workaround to nuget a few hours ago. Update to v1.0.2 and you should be working.

**Reply by plettb** [reply]
Wow!  That's some impressive service! :)

Looks like that did the trick.  Thanks a lot!
